### ISSUE TITLE
bug: NF4 dequant zeros out V projection weights (n=256 k=1536) — causes training NaN

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#233]
## Bug

`apr finetune --quantize-nf4` dequantizes 11 out of 196 weight tensors to all zeros.
All 11 are shape 256x1536 — the V (value) projection weights in GQA attention.

## Evidence (apr trace on yoga)

Pattern repeats for every attention layer:
```
[TRACE] dequant n=256 k=1536 len=393216 nonzero=354629 first5=[-0.037..]  ← K proj (OK)
[TRACE] dequant n=256 k=1536 len=393216 nonzero=0 first5=[0.0, 0.0, ..]  ← V proj (ZERO!)
```

11 zero tensors / 196 total = 5.6% of weights completely zeroed.

## Impact

Zero V weights → zero attention values → softmax(0/0) = NaN → propagates through
all 28 layers → `[CUDA] NaN in forward output` → CPU lm_head fallback → 44 tok/s
instead of expected 2000+ tok/s.

## Diagnostic Chain

1. `apr profile --granular` on Q4K: inference works fine (151 tok/s, grade D, memory bound)
2. `apr finetune --quantize-nf4`: NaN in forward, 0% GPU utilization
3. `apr trace` (grep nonzero=0): 11 V-projection tensors dequant to zeros
4. Same behavior with both Q4K and FP16 models → NF4 runtime dequant is the bug

## Root Cause Hypothesis

The NF4 dequantization kernel handles the K and V projection tensors differently
despite identical shapes (256x1536). Possible causes:
- Off-by-one in the quantization group boundary calculation for V weights
- V weights stored in a different memory layout (row-major vs col-major)
- NF4 codebook lookup produces 0 for the V weight value distribution

## Contract Reference

`apr-training-parity-v1.yaml` H-PARITY-001 FALSIFIED (not model format),
H-PARITY-002 refinement: NF4 dequant is the root cause, specifically V projection.

Refs: PMAT-455, paiml/qwen-train-canary#13
